src/prometheus/registry.zig

```diff
@@ -238,13 +238,12 @@ pub fn Registry(comptime options: RegistryOptions) type {
             if (self.nbMetrics() >= options.max_metrics) return error.TooManyMetrics;
             if (name.len > options.max_name_len) return error.NameTooLong;
 
-            var allocator = self.arena_state.allocator();
-
-            const duped_name = try allocator.dupe(u8, name);
-
             self.mutex.lock();
             defer self.mutex.unlock();
 
+            var allocator = self.arena_state.allocator();
+            const duped_name = try allocator.dupe(u8, name);
```

Comment:
Should either just be `self.arena_state.allocator().dupe(u8, name)`, or some variation of `const allocator = self.arena_state.allocator()` or `const arena = self.arena_state.allocator()`.

Reply:
[ea137f5](https://github.com/Syndica/sig/pull/522/commits/ea137f5d3ee3252bbe8fc2e625d1129e5c310da9)